pandas/core/frame.py

```diff
@@ -6748,6 +6748,22 @@ def drop_duplicates(
         DataFrame or None
             DataFrame with duplicates removed or None if ``inplace=True``.
 
+        Notes
+        -------
+        To handle mutable objects such as list, convert the list column
+        to a tuple before using it in the subset.
```

Comment:
This does not handle all cases. I think the note should start with something more general like:

    This method requires data in the columns to be hashable.